OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -397,42 +456,68 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 						prefix += "|";
 				}
 
-				var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), Game.ModData.VideoLoaders);
-				if (video != null)
+				var fileExtension = Path.GetExtension(filename.ToLowerInvariant());
+				if (allowedSpriteExtensions.Contains(fileExtension))
 				{
-					player = panel.Get<VideoPlayerWidget>("PLAYER");
-					player.Load(prefix + filename);
-					player.DrawOverlay = false;
-					isVideoLoaded = true;
+					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
+					currentFrame = 0;
 
 					if (frameSlider != null)
 					{
-						frameSlider.MaximumValue = (float)player.Video.Frames - 1;
-						frameSlider.Ticks = 0;
+						frameSlider.MaximumValue = (float)currentSprites.Length - 1;
+						frameSlider.Ticks = currentSprites.Length;
 					}
 
-					return true;
+					currentVoxel = null;
 				}
-
-				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vxl")
+				else if (allowedModelExtensions.Contains(fileExtension))
 				{
 					var voxelName = Path.GetFileNameWithoutExtension(filename);
 					currentVoxel = world.ModelCache.GetModel(voxelName);
 					currentSprites = null;
 				}
-				else
+				else if (allowedAudioExtensions.Contains(fileExtension))
 				{
-					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
-					currentFrame = 0;
+					// Mute music so it doesn't interfere with the current asset.
+					MuteSounds();
 
-					if (frameSlider != null)
+					currentAudioStream = Game.ModData.DefaultFileSystem.Open(prefix + filename);
```

Comment:
Presumably we need to dispose any existing instance assigned to `currentAudioStream ` before reassigning?

Reply:
That is already handled by `LoadAsset` (the current method) calling `ClearLoadedAssets` as the first thing it does. (Please correct me if I'm wrong ... after 1625372516 rebases and fixes there's a non-zero possibility I may have messed something up again)